test-app/package.json

```diff
@@ -34,7 +32,12 @@
     "@ember/optional-features": "^2.2.0",
     "@ember/string": "^4.0.0",
     "@ember/test-helpers": "^4.0.4",
+    "@embroider/broccoli-side-watch": "^1.0.1",
+    "@embroider/compat": "^3.4.8",
+    "@embroider/core": "^3.4.8",
```

Comment:
pnpm identified that these had been missing before (yarn classic had been hoisting them from elsewhere which made them available to the test-app. pnpm doesn't allow that, so they have to be explicitly added)